ResearchKit/Common/ORKTaskViewController.m

```diff
@@ -896,7 +898,7 @@ - (void)showViewController:(ORKStepViewController *)viewController goForward:(BO
     if (step.identifier && ![_managedStepIdentifiers.lastObject isEqualToString:step.identifier]) {
         [_managedStepIdentifiers addObject:step.identifier];
     }
-    if ([step isRestorable]) {
+    if ([step isRestorable] && !(viewController.parentReviewStep && viewController.parentReviewStep.isStandalone)) {
```

Comment:
In a standalone use case, the review step contains steps that do not belong to the current task. Restoring a task view controller back to those steps would fail anyway, since `stepWithIdentifier:` and `stepAfterStep:` will return nil. In this situation it is best to restore the task view controller to the review step itself, which is done by encoding `_lastRestorableStepIdentifier`.